src/test/assets/meson/data/meson.build

```diff
@@ -0,0 +1,41 @@
+desktop_file = i18n.merge_file(
```

Comment:
Do you need such sophisticated thing, we can just install_data and copy the file to the right directory, handling translations and validations are imho a bit out of scope of such a test case.